pkgs/build-support/nix-gitignore/default.nix

```diff
@@ -145,10 +170,10 @@ in rec {
       '');
 
   withGitignoreFile = patterns: root:
-    lib.toList patterns ++ [ ".git" ] ++ [(root + "/.gitignore")];
+    toList patterns ++ [ ".git" ] ++ [(root + "/.gitignore")];
 
   withRecursiveGitignoreFile = patterns: root:
-    lib.toList patterns ++ [ ".git" ] ++ [(compileRecursiveGitignore root)];
+    toList patterns ++ [ ".git" ] ++ [(compileRecursiveGitignore root)];
 
   # filterSource derivatives
```

Comment:
`filter` is used below here as a parameter name. There's no actual overlap -- the parameter name wins, not the `let` binding -- but it made me squeamish, so I scoped the let binding for `filter` to the minimum necessary.

Reply:
Hm well, I don't know. It's a tiny function where it's clearly the parameter name. Actually there it should be `predicate`, not `filter`.

Reply:
Oh man, I only saw the "don't open" sign you left just now. I opened it in https://github.com/NixOS/nixpkgs/pull/295946/commits/93ec3ad95e49c1ef9d9e473935d42a2b25092a0b. I think it turned out fine.

Reply:
Please don't @mention in commits, this will spam notifications on merges and such.

Reply:
Change is fine though. Nice work!

Reply:
Let me reword that commit then.